semantic-analysis/src/Control/Carrier/Readline/Haskeline.hs

```diff
@@ -33,11 +34,11 @@ runReadline prefs settings = runInputTWithPrefs prefs (coerce settings) . runM .
 runReadlineWithHistory :: MonadException m => ReadlineC m a -> m a
 runReadlineWithHistory block = do
   homeDir <- liftIO getHomeDirectory
-  prefs <- liftIO $ readPrefs (homeDir </> ".haskeline")
-  let settingsDir = homeDir </> ".local/semantic-core"
+  prefs <- liftIO $ readPrefs (Path.toString (homeDir </> Path.relFile ".haskeline"))
+  let settingsDir = homeDir </> Path.relDir ".local" </> Path.relDir "semantic-core"
       settings = Settings
         { complete = noCompletion
-        , historyFile = Just (settingsDir <> "/repl_history")
+        , historyFile = Just (Path.toString (settingsDir </> Path.relFile "repl_history"))
```

Comment:
As @patrickt will tell you, `pathtype`’s docs are really horribly organized, but the essence here is that I’m using constructors specific to a) whether the component I’m adding is relative or absolute, and b) whether it’s expected to be a directory or a file. Then I produce the actual `FilePath` (a `Prelude`-provided synonym for `String`) with `Path.toString`, since `haskeline` (and indeed the OS) doesn’t know about `pathtype`.